packages/react-store/src/index.ts

```diff
@@ -66,8 +66,8 @@ export function shallow<T>(objA: T, objB: T) {
     return true
   }
 
-  const keysA = Object.keys(objA)
-  if (keysA.length !== Object.keys(objB).length) {
+  const keysA = Reflect.ownKeys(objA)
```

Comment:
I'm realizing this will iterate over non-enumerable properties as well. Do we want that? If not, I can switch this to `Object.getOwnPropertyNames(objA).concat(Object.getOwnPropertySymbols(objA))` instead.

Reply:
I decided this makes the most sense. I updated this to only get enumerable keys and add a test to document that behavior: [c56d79f](https://github.com/TanStack/store/pull/227/commits/c56d79f8394ecead181474c89b60da9430ffe874)